server/blob/blob-cassandra/src/main/java/org/apache/james/blob/cassandra/CassandraBlobsDAO.java

```diff
@@ -223,18 +224,20 @@ public InputStream read(BlobId blobId) {
             Pipe pipe = Pipe.open();
             ConsumerChainer<ByteBuffer> consumer = Throwing.consumer(
                     bytes -> {
-                        pipe.sink().write(bytes);
-                        pipe.sink().close();
+                        try (Pipe.SinkChannel sink = pipe.sink()) {
+                            sink.write(bytes);
+                        }
                     }
             );
             readBytes(blobId)
                     .thenApply(ByteBuffer::wrap)
-                    .thenAccept(
-                            consumer.sneakyThrow()
-                    );
+                    .thenAccept(consumer.sneakyThrow());
             return Channels.newInputStream(pipe.source());
-        } catch (IOException e) {
-            return null;
+        } catch (IOException cause) {
+            LOGGER.error("Failed to convert CompletableFuture<byte[]> to InputStream", cause);
```

Comment:
Usually, we either log or throw because otherwise we end with duplicated log entries for a single error